salt/_modules/metalk8s_kubernetes.py

```diff
@@ -351,3 +354,37 @@ def list_objects(kind, apiVersion, namespace='default', all_namespaces=False,
         raise CommandExecutionError('{}: {!s}'.format(base_msg, exc))
 
     return [obj.to_dict() for obj in result.items]
+
+
+def get_object_digest(object_key=None, checksum='sha256', *args, **kwargs):
+    """
+    Helper to get the digest of one kubernetes object or from a specific key
+    of this object (usefull to get the digest of one config from ConfigMap)
+
+    CLI Examples:
+
+    .. code-block:: bash
+
+        salt-call metalk8s_kubernetes.get_object_digest kind="ConfigMap" apiVersion="v1" name="my-config-map" object_key="config.yaml"
+        salt-call metalk8s_kubernetes.get_object_digest kind="Pod" apiVersion="v1" name="my-pod"
+    """
+    obj = get_object(*args, **kwargs)
+
+    if not obj:
+        raise CommandExecutionError('Unable to find the object')
+
+    if object_key:
+        if not isinstance(object_key, list):
+            object_key = object_key.split(':')
+
+        for key in object_key:
+            try:
+                obj = obj[key]
+            except KeyError:
+                raise CommandExecutionError(
+                    'Unable to find key "{}" in the object'.format(
+                        '.'.join(object_key)
+                    )
+                )
```

Comment:
Can't we use Salt utils for this?

Reply:
Which one ?

Reply:
[This one](https://github.com/saltstack/salt/blob/master/salt/utils/data.py#L785)?

Reply:
Right, I will